File: scripts/check_settings.py
import sqlite3
import os
import json

db_path = os.path.join(os.environ['USERPROFILE'], 'AppData', 'Roaming', 'StudyTube', 'studytube.db')

if not os.path.exists(db_path):
    print(f"Database not found at {db_path}")
    exit(1)

try:
    conn = sqlite3.connect(db_path)
    cur = conn.cursor()
    cur.execute("SELECT config FROM settings WHERE id = 1")
    row = cur.fetchone()
    if row:
        print("Current Settings JSON:")
        print(json.dumps(json.loads(row[0]), indent=2))
    else:
        print("No settings found.")
    conn.close()
except Exception as e:
    print(f"Error: {e}")

File: .agent/scripts/check_settings.js
import Database from 'better-sqlite3';
import path from 'path';
import os from 'os';

const dbPath = path.join(os.homedir(), 'AppData', 'Roaming', 'StudyTube', 'studytube.db');
console.log('Checking DB at:', dbPath);

try {
    const db = new Database(dbPath);
    const row = db.prepare('SELECT config FROM settings WHERE id = 1').get();
    if (row) {
        const config = JSON.parse(row.config);
        console.log('Current Settings:', config);
        console.log('Cloud API URL:', config.cloudApiUrl || 'NOT SET');
    } else {
        console.log('Settings table is empty or record id=1 missing.');
    }
} catch (err) {
    console.error('Error reading DB:', err.message);
}
